STM32F405RGTx/Core/Src/rtos_threads/coulomb_counting_thread.cpp

```diff
@@ -4,51 +4,60 @@
 #include "bms_entry.hpp"
 #include "threads.hpp"
 
-#define MEASUREMENT_PERIODICITY_SECONDS MEASUREMENT_PERIODICITY * 1E-3
-#define NOMINAL_CAP     21600
-#define INIT_SOC        100
-
 RTOSThread CoulombCountingThread::thread;
 float CoulombCountingThread::prev_current;
+uint32_t CoulombCountingThread::prev_time;
+
+osThreadId_t CoulombCountingThread::getThreadId() { return CoulombCountingThread::thread.getThreadId(); }
 
 void CoulombCountingThread::initialize() {
     thread = RTOSThread(
         "coulomb_counting_thread",
         1024,
-        osPriorityNormal,
+        COULOMB_COUNTING_THREAD_PRIORITY,
         runCoulombCounting
     );
 }
 
 float CoulombCountingThread::getCharge() {
```

Comment:
Maybe rename this from get charge to get used charge or something cause that's really what its doing